### PR TITLE
JS: recognize more startswith sanitizers for path-injection queries

### PR DESCRIPTION
**Pull request by erik-krogh**
Gets a TN for CVE-2021-23484 

[An evaluation seems clean](https://github.com/github/codeql-dca-main/tree/data/erik-krogh/pr-7876-ca5f91e__nightly__CustomSuite/reports).

### CODE REVIEW

File: javascript/ql/lib/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll [hunk 1]
```diff
@@ -513,13 +513,24 @@ module TaintedPath {
...
       outcome = startsWith.getPolarity().booleanNot()
       or
       not member = "relative" and
-      e = pathCall.getArgument(0).asExpr() and
+      e = maybeGetJoinArg(pathCall.getArgument(0)).asExpr() and
       outcome = startsWith.getPolarity()
     }
+
+    bindingset[e]
+    private DataFlow::Node maybeGetJoinArg(DataFlow::Node e) {
```
Comment (esbena) [inline javascript/ql/lib/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll:525]: I'm having trouble relating the name and the semantics of this predicate. Could you add a bit of qldoc? Alternative names, based on semantics: `maybeGetLastJoinArg`, `getPathSuffix`?

Comment (erik-krogh) [reply javascript/ql/lib/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll]: I ended up with `maybeGetPathSuffix`.  
I want the `maybe` in there to capture that the predicate also acts as the identity function.

**Review by esbena — DISMISSED** [review]
(do you want to deal with the ql-for-ql alerts in this PR?)

**Comment by erik-krogh** [comment]
> (do you want to deal with the ql-for-ql alerts in this PR?)

(We could use some automatic suggestions on that).

**Review by esbena — APPROVED** [review]

**Pull request merged**